### ISSUE TITLE
Some functions not labeled properly in Android shared libraries

### ISSUE THREAD

**Issue by skywardgaze** [radareorg/cutter#1687]
**Environment information**
* Operating System: Kali 
* Cutter version: 1.8.3 
* File format: Shared Library (.so) 

**Describe the bug**

I've been using Cutter to solve some Crackmes, I ran into this issue with [this one](https://github.com/OWASP/owasp-mstg/blob/master/Crackmes/Android/Level_03/UnCrackable-Level3.apk) specifically. If you unpack or unzip the APK you will find a library in the lib folder called libfoo.so, which has some dynamically linked Java methods. Java methods in native libraries called by Android apps need to be named as such `Java_com_android_interesting_Stuff_doThingsInNativeLibrary`. In libfoo.so, Cutter detects two of the native methods properly but does not detect `Java_sg_vantagepoint_uncrackable3_MainActivity_init`. If I search for the function I can find it, but on the function bar on the left, it is named entry.init0.

**Expected behavior**

Other disassemblers (like Ghidra) preserve the original name of the function when listing functions, I think that Cutter should follow this behavior as well.

**Comment by skywardgaze** [comment]
I also noticed that the pseudocode for the function does not show up as well.

**Comment by pelijah** [comment]
@radare overlapped functions
![изображение](https://user-images.githubusercontent.com/16493991/61821279-70da6d80-ae5f-11e9-9f11-2eda5d7b4fea.png)

**Comment by pelijah** [comment]
I assume that r2 doesn't properly handle call to `__stack_chk_fail`.

**Comment by pelijah** [comment]
Ah, r2 doesn't even know about `__stack_chk_fail`. Something is wrong with loading function signatures.

**Comment by pelijah** [comment]
Yeah r2 doesn't know what signatures to load for Android. I think it should load `types-linux.sdb`.

**Comment by pelijah** [comment]
Lol, there is no `__stack_chk_fail` in `types-android.sdb.txt`.

**Comment by pelijah** [comment]
But function's name is still `entry.init0`
![изображение](https://user-images.githubusercontent.com/16493991/61825111-535dd180-ae68-11e9-82b8-d54319347bbe.png)

**Comment by pelijah** [comment]
Ok, r2 sets flags `entry.init0` and `Java_sg_vantagepoint_uncrackable3_MainActivity_init` at the same offset. On the `aa` step r2 just takes the first flag and creates a function. IDK maybe r2 should prefer non `entry.*` flags for function name.

**Comment by pelijah** [comment]
@radareorg/cutter-maintainers  I think it should be moved to r2 repo.

**Comment by ITAYC0HEN** [comment]
I agree. Is it still reproducible with latest r2?

**Comment by ITAYC0HEN** [comment]
@trufae is this issue tracked on radare2? If no, we will open one to track it

**Comment by pelijah** [comment]
Transfer the issue to r2 repo please.
